components/BookCatalog/CheckInOrOut.tsx

```diff
@@ -21,10 +27,147 @@ const style = {
   padding: '32px 32px 12px 32px'
 };
 
+interface CheckoutPostBody {
+  memberId: string;
+  bookId: string;
+  dueDate: string;
+}
+
+interface CheckoutPatchBody {
+  bookId: string;
+}
+
 export default function CheckInOrOut({ title, CheckInOrOut }) {
-  const [open, setOpen] = React.useState(false);
+  // const currentDate = new Date();
+  // const defaultDueDate = new Date(
+  //   currentDate.getTime() + 3 * 7 * 24 * 60 * 60 * 1000
+  // );
+
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [dueDate, setDueDate] = useState(null);
+  // const [dueDate, setDueDate] = useState<dayjs.Dayjs>(dayjs(new Date()));
+
+  const [barcode, setBarcode] = useState<string>('');
+  const [memberId, setMemberId] = useState<string>('');
+  const [buttonText, setButtonText] = useState<string>(
+    title === 'Check Out' ? 'Check Out' : 'Check In'
+  );
+  const { data, setData } = useContext(BooksContext);
+
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
+  const handleCheckInOrOut = async () => {
+    let response;
+    const url =
+      process.env.NODE_ENV === 'production'
+        ? 'http://localhost:3000'
+        : 'http://localhost:3000';
```

Comment:
You can swap this for the library util, something like `getEnvUrl()`. It's functionally identical, but keeps things a bit DRYer

Reply:
Swapped it out with getEnvUrl()